Utility/Connect-AzureVM.ps1

```diff
@@ -27,103 +27,198 @@
-        $AzureX509cert = Get-AzureCertificate -ServiceName $Using:ServiceName -Thumbprint $winRMCert -ThumbprintAlgorithm sha1
+}
+
+    Configure-AzureWinRMHTTPS -ResourceGroupName $ResourceGroupName -VMName $VMName
 
-        # Add the VM certificate into the LocalMachine
-        if ((Test-Path Cert:\LocalMachine\Root\$winRMCert) -eq $false)
+    $VM = Get-AzureRmVM -ResourceGroupName $ResourceGroupName -Name $VMName
+    if($VM)
+    {
+  
+        $NICs = Get-AzureRmNetworkInterface | Where-Object{$_.VirtualMachine.Id -eq $VM.Id}
+        $IPConfigArray = New-Object System.Collections.ArrayList
+        foreach($Nic in $NICs)
         {
-            Write-Progress "VM certificate is not in local machine certificate store - adding it"
-            $certByteArray = [System.Convert]::fromBase64String($AzureX509cert.Data)
-            $CertToImport = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2 -ArgumentList (,$certByteArray)
-            $store = New-Object System.Security.Cryptography.X509Certificates.X509Store "Root", "LocalMachine"
-            $store.Open([System.Security.Cryptography.X509Certificates.OpenFlags]::ReadWrite)
-            $store.Add($CertToImport)
-            $store.Close()
+            if($Nic.IpConfigurations.LoadBalancerBackendAddressPools)
+            {
+                $Arr = $Nic.IpConfigurations.LoadBalancerBackendAddressPools.id.Split('/')
+                $LoadBalancerNameIndex = $Arr.IndexOf("loadBalancers") + 1                    
+                $LoadBalancer = Get-AzureRmLoadBalancer | Where-Object{$_.Name -eq $Arr[$LoadBalancerNameIndex]}
+                $PublicIpId = $LoadBalancer.FrontendIPConfigurations.PublicIpAddress.Id
+            }
+
+            $PublicIps = New-Object System.Collections.ArrayList
+
+            if($Nic.IpConfigurations.PublicIpAddress.Id)
+            {
+                $PublicIps.Add($Nic.IpConfigurations.PublicIpAddress.Id) | Out-Null
+            }
+
+            if($PublicIpId)
+            {
+                $PublicIps.Add($PublicIpId) | Out-Null
+            }
+
+            foreach($PublicIp in $PublicIps)
+            {
+                $Name = $PublicIp.split('/')[$PublicIp.Split('/').Count - 1]
+                $ResourceGroup = $PublicIp.Split('/')[$PublicIp.Split('/').Indexof("resourceGroups")+1]
+                $PublicIPAddress = Get-AzureRmPublicIpAddress -Name $Name -ResourceGroupName $ResourceGroup | Select-Object -Property Name,ResourceGroupName,Location,PublicIpAllocationMethod,IpAddress
+                $IPConfigArray.Add($PublicIPAddress) | Out-Null
+            }
+        }
+        $Uri = $IPConfigArray | Where-Object{$_.IpAddress -ne $null} | Select-Object -First 1 -Property IpAddress
+
+        if($Uri.IpAddress -ne $null)
+        {               
+            return $Uri.IpAddress.ToString()           
```

Comment:
shouldn't full uri that can just be passed to -connectionuri be returned? why make user construct it themselves?